### PR TITLE
chore!: Remove `deprecated_serialize_commitment` from API

### PR DESCRIPTION
**Pull request by kevaundray**
## Rationale 

TLDR: Imagine you were using a cryptography library which exposed two methods `hash1` and `hash2` where the difference between the two of them is that `hash1` uses keccak256 and `hash2` uses `sha256`. From the callers perspective you get a 32 byte slice regardless, so having both of these in the API does not add any extra functionality. We can therefore remove one.

The existence of `deprecated_serialize_commitment` on the API allows downstream users to serialize a commitment to a 32 byte array. `hash_commitment` holds the same functionality, with the exception that the 32 byte array represents a field element. From the perspective of the verkle trie library, these two methods provide the same functionality.

Having two ways to convert a commitment to a 32 byte array, where they are used in different places in the code allows for footguns and confusions especially when trying to decide which one to use. Using one in place of the other in the public API, makes it seem like one of them is more special than the other.

See #86

### CODE REVIEW

**Comment by kevaundray** [comment]
Reverting this because serialize is used to serialize the root node, mapToScalarField is not reversible unfortunately
